flink-ai-flow/ai_flow/api/ops.py

```diff
@@ -553,151 +426,159 @@ def user_define_operation(
-#                                       ) -> None:
-#     """
-#     Add restart-before control dependency. It means src channel will restart when and only the dependency channel stop.
-#
-#     :param namespace:
-#     :param src: The src channel depended on the dependency channel.
-#     :param dependency: The channel which is the dependency.
-#     :return: None.
-#     """
-#
-#     tmp = RestartBeforeControlEdge(source_node_id=src.node_id,
-#                                    target_node_id=dependency.node_id,
-#                                    namespace=namespace)
-#     _default_ai_graph.add_edge(src.node_id, tmp)
-
-
-def model_version_control_dependency(src: Channel,
-                                     model_name: Text,
-                                     model_version_event_type,
-                                     dependency: Channel,
-                                     namespace: Text = DEFAULT_NAMESPACE
-                                     ) -> None:
+       Add user defined control logic.
+       :param job_name: The job name identify the job.
+       :param namespace: the project name
+       :param event_key: The key of the event.
+       :param event_value: The value of the event.
+       :param event_type: The Name of the event.
+       :param condition: The event condition. Sufficient or Necessary.
+       :param action: The action act on the src channel. Start or Restart.
+       :param life: The life of the event. Once or Repeated.
+       :param value_condition: The event value condition. Equal or Update. Equal means the src channel will start or
+                               restart only when in the condition that the notification service updates a value which
+                               equals to the event value under the specific event key, while update means src channel
+                               will start or restart when in the the condition that the notification service has a update
+                               operation on the event key which event value belongs to.
+       :param sender: The event sender identity. If sender is None, the sender will be dependency.
+       :return:None.
+       """
+    control_edge = ControlEdge(head=job_name,
+                               event_key=event_key,
+                               event_value=event_value,
+                               event_type=event_type,
+                               condition=condition,
+                               action=action,
+                               life=life,
+                               value_condition=value_condition,
+                               namespace=namespace,
+                               sender=sender
+                               )
+    default_graph().add_edge(job_name, control_edge)
+
+
+def action_on_model_version_event(job_name: Text,
+                                  model_name: Text,
+                                  model_version_event_type: Text,
```

Comment:
What does `model_version_event_type` mean? Do you mean the `model_version_event_type` represent the `status` of the model version? What's more, there's no comment for this parameter.